### PR TITLE
Fix/1391 a11y text on badge

### PR DESCRIPTION
**Pull request by GCHQ-Developer-847**
## Summary of the changes

Adjusted screen reader behaviour on badge component. Includes some changes which are a bit out of scope but felt necessary to add to resolve a few issues I have come across.

I have created a [StackBlitz](https://stackblitz.com/edit/stackblitz-starters-9cjbv1?file=src%2FApp.tsx) which helps to demonstrate the existing screen reader behaviour of badges, and shows the following issues:
- If there is no accessible label, the user is not aware that a badge is present (although the text label _is_ automatically read out on cards and tabs, but with no comma / break between text)
- Repetition of accessible labels, e.g. on side navigation and cards
- Aria-labels and accessible labels sometimes override each other

See [my research in a comment on the ticket](https://github.com/mi6/ic-ui-kit/issues/1391#issuecomment-2010305102) for my reasoning for certain changes (e.g. including the solution for the main issue raised in the ticket description).

----------

**How to test these changes**

The easiest way to test is probably to copy and paste the code below into a story so you can see all possible parent components for badges (with all combinations of accessible + text label) on one page, and then check using a screen reader.

<details>

<summary>VIEW CODE</summary>

```js
  /* Five examples of each possible parent component for a badge are displayed.
      This list shows what is different with the badge on each one. */
  <p>1. Text label, no accessible label<br />
      2. No text label (with type="dot"), no accessible label<br />
      3. Text label, accessible label<br />
      4. No text label (with type="dot"), accessible label<br />
      5. Aria-label on parent, accessible label</p>
      <br />
      <div style="display: flex; flex-direction: column; gap: 16px;">
        <div>
          <ic-button variant="secondary">
            <ic-badge text-label="1" slot="badge"></ic-badge>
            Default
          </ic-button>
          <ic-button variant="secondary">
            <ic-badge slot="badge" type="dot"></ic-badge>
            Default
          </ic-button>
          <ic-button variant="secondary">
            <ic-badge text-label="1" slot="badge" accessible-label="1 notification found"></ic-badge>
            Default
          </ic-button>
          <ic-button variant="secondary">
            <ic-badge slot="badge" type="dot" accessible-label="Notifications found"></ic-badge>
            Default
          </ic-button>
          <ic-button variant="secondary" aria-label="Aria label for button">
            <ic-badge text-label="1" slot="badge" accessible-label="1 notification found"></ic-badge>
            Default
          </ic-button>
        </div>
        <div>
          <ic-button variant="icon" aria-label="refresh"
            >
            <ic-badge text-label="1" slot="badge"></ic-badge>
            <svg
              xmlns="http://www.w3.org/2000/svg"
              height="24px"
              viewBox="0 0 24 24"
              width="24px"
              fill="#000000"
            >
              <path d="M0 0h24v24H0V0z" fill="none" />
              <path
                d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
              /></svg
          ></ic-button>
          <ic-button variant="icon" aria-label="refresh"
            >
            <ic-badge slot="badge" type="dot"></ic-badge>
            <svg
              xmlns="http://www.w3.org/2000/svg"
              height="24px"
              viewBox="0 0 24 24"
              width="24px"
              fill="#000000"
            >
              <path d="M0 0h24v24H0V0z" fill="none" />
              <path
                d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
              /></svg
          ></ic-button>
          <ic-button variant="icon" aria-label="refresh"
            >
            <ic-badge text-label="1" slot="badge" accessible-label="1 notification found"></ic-badge>
            <svg
              xmlns="http://www.w3.org/2000/svg"
              height="24px"
              viewBox="0 0 24 24"
              width="24px"
              fill="#000000"
            >
              <path d="M0 0h24v24H0V0z" fill="none" />
              <path
                d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
              /></svg
          ></ic-button>
          <ic-button variant="icon" aria-label="refresh"
            >
            <ic-badge slot="badge" type="dot" accessible-label="Notifications found"></ic-badge>
            <svg
              xmlns="http://www.w3.org/2000/svg"
              height="24px"
              viewBox="0 0 24 24"
              width="24px"
              fill="#000000"
            >
              <path d="M0 0h24v24H0V0z" fill="none" />
              <path
                d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
              /></svg
          ></ic-button>
          <ic-button variant="icon" aria-label="Aria label for button"
            >
            <ic-badge text-label="1" slot="badge" accessible-label="1 notification found"></ic-badge>
            <svg
              xmlns="http://www.w3.org/2000/svg"
              height="24px"
              viewBox="0 0 24 24"
              width="24px"
              fill="#000000"
            >
              <path d="M0 0h24v24H0V0z" fill="none" />
              <path
                d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
              /></svg
          ></ic-button>
        </div>
        <div>
          <ic-chip label="Default" dismissible="true">
            <ic-badge
              text-label="1"
              slot="badge"
              position="near"
            ></ic-badge>
          </ic-chip>
          <ic-chip label="Default" dismissible="true">
            <ic-badge
              slot="badge"
              position="near"
              type="dot"
            ></ic-badge>
          </ic-chip>
          <ic-chip label="Default" dismissible="true">
            <ic-badge
              text-label="1"
              slot="badge"
              position="near"
              accessible-label="1 notification found"
            ></ic-badge>
          </ic-chip>
          <ic-chip label="Default" dismissible="true">
            <ic-badge
              slot="badge"
              position="near"
              type="dot"
              accessible-label="Notifications found"
            ></ic-badge>
          </ic-chip>
          <ic-chip label="Default" dismissible="true" aria-label="Aria label for chip">
            <ic-badge
              text-label="1"
              slot="badge"
              position="near"
              accessible-label="1 notification found"
            ></ic-badge>
          </ic-chip>
        </div>
        <div>
          <ic-card heading="This is the card heading" clickable="true" href="https://www.google.com">
            <ic-badge text-label="1" slot="badge"></ic-badge>
          </ic-card>
          <ic-card heading="This is the card heading" clickable="true" href="https://www.google.com">
            <ic-badge slot="badge" type="dot"></ic-badge>
          </ic-card>
          <ic-card heading="This is the card heading" clickable="true" href="https://www.google.com">
            <ic-badge text-label="1" slot="badge" accessible-label="1 notification found"></ic-badge>
          </ic-card>
          <ic-card heading="This is the card heading" clickable="true" href="https://www.google.com">
            <ic-badge slot="badge" type="dot" accessible-label="Notifications found"></ic-badge>
          </ic-card>
          <ic-card heading="This is the card heading" clickable="true" href="https://www.google.com" aria-label="Aria label for card">
            <ic-badge text-label="1" slot="badge" accessible-label="1 notification found"></ic-badge>
          </ic-card>
        </div>
        <div>
          <ic-top-navigation
          app-title="Application Name">
            <svg
              slot="app-icon"
              xmlns="http://www.w3.org/2000/svg"
              height="24px"
              viewBox="0 0 24 24"
              width="24px"
              fill="#000000"
            >
              <path d="M0 0h24v24H0V0z" fill="none" />
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm-5.5-2.5l7.51-3.49L17.5 6.5 9.99 9.99 6.5 17.5zm5.5-6.6c.61 0 1.1.49 1.1 1.1s-.49 1.1-1.1 1.1-1.1-.49-1.1-1.1.49-1.1 1.1-1.1z"
              />
            </svg>
            <ic-navigation-button
              label="Button one"
              slot="buttons"
            >
              <svg
                slot="icon"
                xmlns="http://www.w3.org/2000/svg"
                height="24px"
                viewBox="0 0 24 24"
                width="24px"
                fill="#000000"
              >
                <path d="M0 0h24v24H0V0z" fill="none" />
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm-5.5-2.5l7.51-3.49L17.5 6.5 9.99 9.99 6.5 17.5zm5.5-6.6c.61 0 1.1.49 1.1 1.1s-.49 1.1-1.1 1.1-1.1-.49-1.1-1.1.49-1.1 1.1-1.1z"
                />
              </svg>
              <ic-badge text-label="1" slot="badge"></ic-badge>
            </ic-navigation-button>
            <ic-navigation-button
              label="Button two"
              slot="buttons"
            >
              <svg
                slot="icon"
                xmlns="http://www.w3.org/2000/svg"
                height="24px"
                viewBox="0 0 24 24"
                width="24px"
                fill="#000000"
              >
                <path d="M0 0h24v24H0V0z" fill="none" />
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm-5.5-2.5l7.51-3.49L17.5 6.5 9.99 9.99 6.5 17.5zm5.5-6.6c.61 0 1.1.49 1.1 1.1s-.49 1.1-1.1 1.1-1.1-.49-1.1-1.1.49-1.1 1.1-1.1z"
                />
              </svg>
              <ic-badge slot="badge" type="dot"></ic-badge>
            </ic-navigation-button>
            <ic-navigation-button
              label="Button three"
              slot="buttons"
            >
              <svg
                slot="icon"
                xmlns="http://www.w3.org/2000/svg"
                height="24px"
                viewBox="0 0 24 24"
                width="24px"
                fill="#000000"
              >
                <path d="M0 0h24v24H0V0z" fill="none" />
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm-5.5-2.5l7.51-3.49L17.5 6.5 9.99 9.99 6.5 17.5zm5.5-6.6c.61 0 1.1.49 1.1 1.1s-.49 1.1-1.1 1.1-1.1-.49-1.1-1.1.49-1.1 1.1-1.1z"
                />
              </svg>
              <ic-badge text-label="1" slot="badge" accessible-label="1 notification found"></ic-badge>
            </ic-navigation-button>
            <ic-navigation-button
              label="Button four"
              slot="buttons"
            >
              <svg
                slot="icon"
                xmlns="http://www.w3.org/2000/svg"
                height="24px"
                viewBox="0 0 24 24"
                width="24px"
                fill="#000000"
              >
                <path d="M0 0h24v24H0V0z" fill="none" />
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm-5.5-2.5l7.51-3.49L17.5 6.5 9.99 9.99 6.5 17.5zm5.5-6.6c.61 0 1.1.49 1.1 1.1s-.49 1.1-1.1 1.1-1.1-.49-1.1-1.1.49-1.1 1.1-1.1z"
                />
              </svg>
              <ic-badge slot="badge" type="dot" accessible-label="Notifications found"></ic-badge>
            </ic-navigation-button>
            <ic-navigation-button
              label="Button five"
              slot="buttons"
              aria-label="Aria label for navigation button"
            >
              <svg
                slot="icon"
                xmlns="http://www.w3.org/2000/svg"
                height="24px"
                viewBox="0 0 24 24"
                width="24px"
                fill="#000000"
              >
                <path d="M0 0h24v24H0V0z" fill="none" />
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm-5.5-2.5l7.51-3.49L17.5 6.5 9.99 9.99 6.5 17.5zm5.5-6.6c.61 0 1.1.49 1.1 1.1s-.49 1.1-1.1 1.1-1.1-.49-1.1-1.1.49-1.1 1.1-1.1z"
                />
              </svg>
              <ic-badge text-label="1" slot="badge" accessible-label="1 notification found"></ic-badge>
            </ic-navigation-button>
          </ic-top-navigation>
        </div>
        <div style="display:flex; height:100%;">
          <ic-side-navigation
            app-title="Application Name"
            version="v0.0.0"
            status="BETA"
          >
            <svg
              slot="app-icon"
              xmlns="http://www.w3.org/2000/svg"
              height="24px"
              viewBox="0 0 24 24"
              width="24px"
              fill="#000000"
            >
              <path d="M0 0h24v24H0V0z" fill="none" />
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm-5.5-2.5l7.51-3.49L17.5 6.5 9.99 9.99 6.5 17.5zm5.5-6.6c.61 0 1.1.49 1.1 1.1s-.49 1.1-1.1 1.1-1.1-.49-1.1-1.1.49-1.1 1.1-1.1z"
              />
            </svg>
            <ic-navigation-item label="Item one" href="/" slot="primary-navigation">
              <ic-badge text-label="1" slot="badge"></ic-badge>
              <svg
                slot="icon"
                width="24"
                height="24"
                viewBox="0 0 24 24"
                fill="none"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M23 8C23 9.1 22.1 10 21 10C20.82 10 20.65 9.98 20.49 9.93L16.93 13.48C16.98 13.64 17 13.82 17 14C17 15.1 16.1 16 15 16C13.9 16 13 15.1 13 14C13 13.82 13.02 13.64 13.07 13.48L10.52 10.93C10.36 10.98 10.18 11 10 11C9.82 11 9.64 10.98 9.48 10.93L4.93 15.49C4.98 15.65 5 15.82 5 16C5 17.1 4.1 18 3 18C1.9 18 1 17.1 1 16C1 14.9 1.9 14 3 14C3.18 14 3.35 14.02 3.51 14.07L8.07 9.52C8.02 9.36 8 9.18 8 9C8 7.9 8.9 7 10 7C11.1 7 12 7.9 12 9C12 9.18 11.98 9.36 11.93 9.52L14.48 12.07C14.64 12.02 14.82 12 15 12C15.18 12 15.36 12.02 15.52 12.07L19.07 8.51C19.02 8.35 19 8.18 19 8C19 6.9 19.9 6 21 6C22.1 6 23 6.9 23 8Z"
                  fill="currentColor"
                />
              </svg>
            </ic-navigation-item>
            <ic-navigation-item label="Item two" href="/" slot="primary-navigation">
              <ic-badge slot="badge" type="dot" position="far"></ic-badge>
              <svg
                slot="icon"
                width="24"
                height="24"
                viewBox="0 0 24 24"
                fill="none"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M23 8C23 9.1 22.1 10 21 10C20.82 10 20.65 9.98 20.49 9.93L16.93 13.48C16.98 13.64 17 13.82 17 14C17 15.1 16.1 16 15 16C13.9 16 13 15.1 13 14C13 13.82 13.02 13.64 13.07 13.48L10.52 10.93C10.36 10.98 10.18 11 10 11C9.82 11 9.64 10.98 9.48 10.93L4.93 15.49C4.98 15.65 5 15.82 5 16C5 17.1 4.1 18 3 18C1.9 18 1 17.1 1 16C1 14.9 1.9 14 3 14C3.18 14 3.35 14.02 3.51 14.07L8.07 9.52C8.02 9.36 8 9.18 8 9C8 7.9 8.9 7 10 7C11.1 7 12 7.9 12 9C12 9.18 11.98 9.36 11.93 9.52L14.48 12.07C14.64 12.02 14.82 12 15 12C15.18 12 15.36 12.02 15.52 12.07L19.07 8.51C19.02 8.35 19 8.18 19 8C19 6.9 19.9 6 21 6C22.1 6 23 6.9 23 8Z"
                  fill="currentColor"
                />
              </svg>
            </ic-navigation-item>
            <ic-navigation-item label="Item three" href="/" slot="primary-navigation">
              <ic-badge text-label="1" slot="badge" accessible-label="1 notification found"></ic-badge>
              <svg
                slot="icon"
                width="24"
                height="24"
                viewBox="0 0 24 24"
                fill="none"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M23 8C23 9.1 22.1 10 21 10C20.82 10 20.65 9.98 20.49 9.93L16.93 13.48C16.98 13.64 17 13.82 17 14C17 15.1 16.1 16 15 16C13.9 16 13 15.1 13 14C13 13.82 13.02 13.64 13.07 13.48L10.52 10.93C10.36 10.98 10.18 11 10 11C9.82 11 9.64 10.98 9.48 10.93L4.93 15.49C4.98 15.65 5 15.82 5 16C5 17.1 4.1 18 3 18C1.9 18 1 17.1 1 16C1 14.9 1.9 14 3 14C3.18 14 3.35 14.02 3.51 14.07L8.07 9.52C8.02 9.36 8 9.18 8 9C8 7.9 8.9 7 10 7C11.1 7 12 7.9 12 9C12 9.18 11.98 9.36 11.93 9.52L14.48 12.07C14.64 12.02 14.82 12 15 12C15.18 12 15.36 12.02 15.52 12.07L19.07 8.51C19.02 8.35 19 8.18 19 8C19 6.9 19.9 6 21 6C22.1 6 23 6.9 23 8Z"
                  fill="currentColor"
                />
              </svg>
            </ic-navigation-item>
            <ic-navigation-item label="Item four" href="/" slot="primary-navigation">
              <ic-badge slot="badge" type="dot" position="far" accessible-label="Notifications found"></ic-badge>
              <svg
                slot="icon"
                width="24"
                height="24"
                viewBox="0 0 24 24"
                fill="none"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M23 8C23 9.1 22.1 10 21 10C20.82 10 20.65 9.98 20.49 9.93L16.93 13.48C16.98 13.64 17 13.82 17 14C17 15.1 16.1 16 15 16C13.9 16 13 15.1 13 14C13 13.82 13.02 13.64 13.07 13.48L10.52 10.93C10.36 10.98 10.18 11 10 11C9.82 11 9.64 10.98 9.48 10.93L4.93 15.49C4.98 15.65 5 15.82 5 16C5 17.1 4.1 18 3 18C1.9 18 1 17.1 1 16C1 14.9 1.9 14 3 14C3.18 14 3.35 14.02 3.51 14.07L8.07 9.52C8.02 9.36 8 9.18 8 9C8 7.9 8.9 7 10 7C11.1 7 12 7.9 12 9C12 9.18 11.98 9.36 11.93 9.52L14.48 12.07C14.64 12.02 14.82 12 15 12C15.18 12 15.36 12.02 15.52 12.07L19.07 8.51C19.02 8.35 19 8.18 19 8C19 6.9 19.9 6 21 6C22.1 6 23 6.9 23 8Z"
                  fill="currentColor"
                />
              </svg>
            </ic-navigation-item>
            <ic-navigation-item label="Item five" href="/" slot="primary-navigation" aria-label="Aria label for navigation item">
              <ic-badge text-label="1" slot="badge" accessible-label="1 notification found"></ic-badge>
              <svg
                slot="icon"
                width="24"
                height="24"
                viewBox="0 0 24 24"
                fill="none"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M23 8C23 9.1 22.1 10 21 10C20.82 10 20.65 9.98 20.49 9.93L16.93 13.48C16.98 13.64 17 13.82 17 14C17 15.1 16.1 16 15 16C13.9 16 13 15.1 13 14C13 13.82 13.02 13.64 13.07 13.48L10.52 10.93C10.36 10.98 10.18 11 10 11C9.82 11 9.64 10.98 9.48 10.93L4.93 15.49C4.98 15.65 5 15.82 5 16C5 17.1 4.1 18 3 18C1.9 18 1 17.1 1 16C1 14.9 1.9 14 3 14C3.18 14 3.35 14.02 3.51 14.07L8.07 9.52C8.02 9.36 8 9.18 8 9C8 7.9 8.9 7 10 7C11.1 7 12 7.9 12 9C12 9.18 11.98 9.36 11.93 9.52L14.48 12.07C14.64 12.02 14.82 12 15 12C15.18 12 15.36 12.02 15.52 12.07L19.07 8.51C19.02 8.35 19 8.18 19 8C19 6.9 19.9 6 21 6C22.1 6 23 6.9 23 8Z"
                  fill="currentColor"
                />
              </svg>
            </ic-navigation-item>
          </ic-side-navigation>
        </div>
        <div>
          <ic-tab-context>
            <ic-tab-group label="Example tab group">
              <ic-tab>One<ic-badge text-label="1" slot="badge" position="inline"></ic-badge></ic-tab>
              <ic-tab>Two<ic-badge slot="badge" type="dot" position="inline"></ic-badge></ic-tab>
              <ic-tab>Three<ic-badge text-label="1" slot="badge" accessible-label="1 notification found" position="inline"></ic-tab>
              <ic-tab>Four<ic-badge slot="badge" type="dot" accessible-label="Notifications found" position="inline"></ic-badge></ic-tab>
              <ic-tab aria-label="Aria label for tab">Five<ic-badge text-label="1" slot="badge" accessible-label="1 notification found"
   position="inline"></ic-tab>
            </ic-tab-group>
            <ic-tab-panel>Tab one</ic-tab-panel>
            <ic-tab-panel>Tab two</ic-tab-panel>
            <ic-tab-panel>Tab three</ic-tab-panel>
            <ic-tab-panel>Tab four</ic-tab-panel>
            <ic-tab-panel>Tab five</ic-tab-panel>
          </ic-tab-context>
        </div>
        <div>
          <ic-toggle-button label="Toggle">
            <ic-badge text-label="1" slot="badge"></ic-badge>
          </ic-toggle-button>
          <ic-toggle-button label="Toggle">
            <ic-badge slot="badge" type="dot"></ic-badge>
          </ic-toggle-button>
          <ic-toggle-button label="Toggle">
            <ic-badge text-label="1" slot="badge" accessible-label="1 notification found"></ic-badge>
          </ic-toggle-button>
          <ic-toggle-button label="Toggle">
            <ic-badge slot="badge" type="dot" accessible-label="Notifications found"></ic-badge>
          </ic-toggle-button>
          <ic-toggle-button label="Toggle" aria-label="Aria label for toggle button">
            <ic-badge text-label="1" slot="badge" accessible-label="1 notification found"></ic-badge>
          </ic-toggle-button>
        </div>
      </div>
```

</details>

The other option is to use a screen reader on any story that has a badge in it (I've added some `type="dot"` examples) - so that’s any of the badge component stories, the “With badge” card story, the “Default” side navigation story, the “With badge” toggle button story, and the “With icons” top navigation story. 

## Related issue
#1391 (and kind of #1419)

### General 

- [x] Changes to docs package checked and committed.
- [x] All acceptance criteria reviewed and met. 

### Testing

- [x] Relevant unit tests and visual regression tests added. 
- [x] Visual testing against Figma component specification completed. 

### Accessibility 

- [x] Accessibility Insights FastPass performed.
- [x] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [x] Manual screen reader testing performed using NVDA and VoiceOver. 
- [x] Manual keyboard testing for keyboard controls and logical focus order. 
- [x] Correct roles used and ARIA attributes used correctly where required. 
- [x] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 

### Resize/zoom behaviour 

- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Screen magnifier used with no issues. 
- [x] Text resized to 200% with no loss of content.
- [x] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.

### System modes

- [x] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [x] Windows High Contrast mode tested with no loss of content.
- [x] System light and dark mode tested with no loss of content.
- [x] Browser support tested (Chrome, Safari, Firefox and Edge). 

### Testing content extremes

- [x] Min/max content examples tested with no loss of content or overflow. 
- [x] All prop combinations work without issue. 
- [x] Tested for FOUC (Flash of Unstyled Content) in both SSR (Server-Side Rendering) and SSG (Static Site Generation) settings.

### CODE REVIEW

**Comment by GCHQ-Developer-465** [comment]
Looks good! There are some merge conflicts that need addressing :)

File: packages/react/src/stories/ic-side-navigation.stories.mdx [hunk 1]
```diff
@@ -661,6 +677,7 @@ export const DailyTippers = () => (
           </svg>
         </NavLink>
         <IcNavigationItem slot="primary-navigation">
+          <IcBadge text-label="1" slot="badge" variant="light" />
```
Comment (GCHQ-Developer-741) [inline packages/react/src/stories/ic-side-navigation.stories.mdx:680]:
```suggestion
          <IcBadge textLabel="1" slot="badge" variant="light" />
```

File: packages/react/src/stories/ic-badge.stories.mdx [hunk 2]
```diff
@@ -829,7 +829,7 @@ export const showHideBadge = () => {
           type="dot"
           slot="badge"
           position="near"
-          accessible-label="1 notification found"
+          accessible-label="Notifications found"
```
Comment (GCHQ-Developer-741) [inline packages/react/src/stories/ic-badge.stories.mdx:832]:
```suggestion
          accessibleLabel="Notifications found"
```

File: packages/react/src/stories/ic-side-navigation.stories.mdx [hunk 3]
```diff
@@ -42,13 +42,29 @@ import { NavLink, MemoryRouter, Route, Routes } from "react-router-dom";
         <path d="M0 0h24v24H0V0z" fill="none" />
         <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm-5.5-2.5l7.51-3.49L17.5 6.5 9.99 9.99 6.5 17.5zm5.5-6.6c.61 0 1.1.49 1.1 1.1s-.49 1.1-1.1 1.1-1.1-.49-1.1-1.1.49-1.1 1.1-1.1z" />
       </svg>
+      <IcNavigationItem slot="primary-navigation" href="/" label="Item 1">
+        <IcBadge text-label="1" slot="badge" variant="light" position="far" />
```
Comment (GCHQ-Developer-741) [inline packages/react/src/stories/ic-side-navigation.stories.mdx:46]:
```suggestion
        <IcBadge textLabel="1" slot="badge" variant="light" position="far" />
```

**Review by GCHQ-Developer-741 — APPROVED** [review]

**Review by GCHQ-Developer-465 — APPROVED** [review]

**Pull request merged**